modules/apps/27-interchain-accounts/controller/ibc_middleware_test.go

```diff
@@ -960,22 +962,31 @@ func (suite *InterchainAccountsTestSuite) TestOnChanUpgradeAck() {
 func (suite *InterchainAccountsTestSuite) TestOnChanUpgradeOpen() {
 	var (
 		path                *ibctesting.Path
-		isNilApp            bool
 		counterpartyVersion string
 	)
 
 	testCases := []struct {
 		name     string
 		malleate func()
+		expPanic func() error
```

Comment:
could just make this an `error` it seems, no need for a func if it unconditionally returns an `err`.

Reply:
I let `expPanic` be a `func` because `channeltypes.ErrChannelNotFound` needs `portId` and `channelId` as inputs. If I set up the path variable outside the loop, the test case "failure: connection not found" will affect the test cases that follow.

Reply:
ah that's a great point, missed it! See: https://github.com/cosmos/ibc-go/blob/0d5d684bcd4e4e3561fdfde5571bbda072d3c8dd/modules/apps/27-interchain-accounts/controller/ibc_middleware_test.go#L33 to get around it :smile:

Reply:
Thanks for pointing that out for me :sweat_smile: